xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/hibernate/query/HqlQueryExecutor.java

```diff
@@ -103,32 +112,39 @@ public void initialize() throws InitializationException
         configuration.addInputStream(Util.getResourceAsStream(MAPPING_PATH));
     }
 
-    private Set<String> getAllowedNamedQueries()
+    private Set<String> getSafeNamedQueries()
     {
-        if (this.allowedNamedQueries == null) {
+        if (this.safeNamedQueries == null) {
             synchronized (this) {
-                if (this.allowedNamedQueries == null) {
-                    this.allowedNamedQueries = new HashSet<>();
+                if (this.safeNamedQueries == null) {
+                    this.safeNamedQueries = new HashSet<>();
```

Comment:
This property should only be set at the end of the method, to avoid that another thread reads the set while it is initialized (the same problem existed already before, SonarQube finds this problem).

Reply:
It indeed does not have much to do with this pull request, but I guess I can squeeze it anyway.